### DISCUSSION TITLE
Read Allen-Bradley EtherNet/IP Encoder

### DISCUSSION THREAD

**Discussion by MAD-Thrustm** [ottowayi/pycomm3#315] [Q&A]
I'm seeking a method to retrieve the value of 842E-MIP-xxxEtherNet/IP. Any assistance would be greatly appreciated.
Here's the code snippet which return:
{'encap_protocol_version': 1, 'ip_address': '192.168.1.112', 'vendor': 'Rockwell Automation/Allen-Bradley', 'product_type': 'Encoder', 'product_code': 2, 'revision': {'major': 1, 'minor': 2}, 'status': b'5\x00', 'serial': '800d5f94', 'product_name': '842E-M', 'state': 3}

from pycomm3 import CIPDriver
if __name__ == "__main__":
    print(CIPDriver.list_identity('192.168.1.112'))

**Comment by Colt-H** [comment]
This is using values from the manual for the device, I don't have one to test it with.

```
#Class/Instance/Attribute from Rockwell Document 842e-um001_-en-p.pdf

with CIPDriver("10.0.0.1") as encoder:
    position = encoder.generic_message(
            service=Services.get_attribute_single,
            class_code=ClassCode.position_sensor,
            instance=b"\x01",
            attribute=10,
            data_type='DINT'
    )
    print(position)
```
